coop_membership/models/shift_mail_registration.py

```diff
@@ -10,17 +10,18 @@ class ShiftMailRegistration(models.Model):
 
     @api.multi
     def execute(self):
-        shift_id = self.registration_id.shift_id
-        shift_holiday = shift_id.long_holiday_id or shift_id.single_holiday_id
-        if shift_holiday and not shift_holiday.send_email_reminder:
-            return
-        if shift_holiday and shift_holiday.reminder_template_id:
-            if (
-                self.registration_id.state in ['open', 'done']
-                and not self.mail_sent
-            ):
-                shift_holiday.reminder_template_id.send_mail(
-                    self.registration_id.id)
-                self.write({'mail_sent': True})
-            return
+        for rec in self:
+            shift_id = rec.registration_id.shift_id
+            shift_holiday = shift_id.long_holiday_id or shift_id.single_holiday_id
+            if shift_holiday and not shift_holiday.send_email_reminder:
+                return
+            if shift_holiday and shift_holiday.reminder_template_id:
+                if (
+                    rec.registration_id.state in ['open', 'done']
+                    and not rec.mail_sent
+                ):
+                    shift_holiday.reminder_template_id.send_mail(
+                        rec.registration_id.id)
+                    rec.write({'mail_sent': True})
+                return
         return super(ShiftMailRegistration, self).execute()
```

Comment:
This won't work.
You need to refactor the function.

If you `return` while processing the recorset, it won't work.

You may need to refactor this by filtering in the recorset and coming up with:

- records_to_ignore
- records_to_send_email

and processing them accordingly.